extensions/websocket/01_websocket.js

```diff
@@ -161,148 +273,122 @@
+      webidl.assertBranded(this, WebSocket);
+      const prefix = "Failed to execute 'send' on 'WebSocket'";
 
-      if (this.#readyState != OPEN) {
-        throw Error("readyState not OPEN");
+      webidl.requiredArguments("WebSocket.send", arguments.length, 1, {
+        prefix,
+      });
+      data = webidl.converters.WebSocketSend(data, {
+        prefix,
+        context: "Argument 1",
+      });
+
+      if (this[_readyState] !== OPEN) {
+        throw new DOMException("readyState not OPEN", "InvalidStateError");
       }
 
       const sendTypedArray = (ta) => {
-        this.#bufferedAmount += ta.size;
+        this[_bufferedAmount] += ta.byteLength;
         core.opAsync("op_ws_send", {
-          rid: this.#rid,
+          rid: this[_rid],
           kind: "binary",
         }, ta).then(() => {
-          this.#bufferedAmount -= ta.size;
+          this[_bufferedAmount] -= ta.byteLength;
         });
       };
 
       if (data instanceof Blob) {
         data.slice().arrayBuffer().then((ab) =>
           sendTypedArray(new DataView(ab))
         );
-      } else if (
-        data instanceof Int8Array || data instanceof Int16Array ||
-        data instanceof Int32Array || data instanceof Uint8Array ||
-        data instanceof Uint16Array || data instanceof Uint32Array ||
-        data instanceof Uint8ClampedArray || data instanceof Float32Array ||
-        data instanceof Float64Array || data instanceof DataView
-      ) {
+      } else if (ArrayBuffer.isView(data)) {
         sendTypedArray(data);
       } else if (data instanceof ArrayBuffer) {
         sendTypedArray(new DataView(data));
       } else {
         const string = String(data);
         const d = core.encode(string);
-        this.#bufferedAmount += d.size;
+        this[_bufferedAmount] += d.byteLength;
         core.opAsync("op_ws_send", {
-          rid: this.#rid,
+          rid: this[_rid],
           kind: "text",
           text: string,
         }).then(() => {
-          this.#bufferedAmount -= d.size;
+          this[_bufferedAmount] -= d.byteLength;
         });
       }
     }
 
-    close(code, reason) {
-      if (code && !(code === 1000 || (3000 <= code && code < 5000))) {
-        throw new DOMException(
-          "The close code must be either 1000 or in the range of 3000 to 4999.",
-          "NotSupportedError",
-        );
+    close(code = undefined, reason = undefined) {
+      webidl.assertBranded(this, WebSocket);
+      const prefix = "Failed to execute 'close' on 'WebSocket'";
+
+      if (code !== undefined) {
+        code = webidl.converters["unsigned short"](code, {
+          prefix,
+          clamp: true,
+          context: "Argument 1",
+        });
+
+        if (!(code === 1000 || (3000 <= code && code < 5000))) {
+          throw new DOMException(
+            "The close code must be either 1000 or in the range of 3000 to 4999.",
+            "InvalidAccessError",
+          );
+        }
```

Comment:
This and the `if (core.encode(reason).byteLength > 123)` have to happen after the two idl converters. Ordering is incorrect otherwise.